### ISSUE TITLE
[JAVA] Ability to construct a Tensor from a GPU memory pointer

### ISSUE THREAD

**Issue by balenamiaa** [microsoft/onnxruntime#20966]
### Describe the issue

The C# API:
```csharp
        var dataPtr = cudaResource.GetMappedPointer();
        var ortValue = OrtValue.CreateTensorValueWithData(MemoryInfo, tensorElementType, Shape, dataPtr, 3 * width * height * sizeof(float));
```

It allows for creating a tensor from a raw GPU pointer, thus avoiding copying the data to CPU, which is important in cases where even 1 extra millisecond of latency matters. No such API is exposed on the JVM side of things. I can get around it currently by doing FFI with Project Panama with the c-api for the onnxruntime, but it should probably be exposed through JNI and have a Java API for it.

### To reproduce

~

### Platform

Windows

### OS Version

~

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

~

### ONNX Runtime API

Java

### Architecture

X64

### Execution Provider

CUDA, TensorRT

**Comment by Craigacp** [comment]
If we expose direct creation of GPU tensors we'd have to have a data copy somewhere. I'm currently not sure where is the best place for that copy, and if the GPU tensors should be different in the type system somehow to make it clear that they don't have direct access to the memory.

Is the use case in a pipeline of models all resident on the GPU?

**Comment by balenamiaa** [comment]
Yup, the models are on the GPU. In my case, it's not a direction creation of GPU tensor, but rather wrapping a raw resource, a D3D11 texture, converted into a CUDA resource with [CUDA's Direct3D 11 Interoperability](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__D3D11.html). [CreateTensorWithDataAsOrtValue](https://onnxruntime.ai/docs/api/c/struct_ort_api.html#a2aad3ccd68c66d0b38bdb966467d9324) is used for this. I don't think the CSharp API has any distinction between an OrtValue that is backed by a GPU buffer or a CPU buffer. In fact, calling GetTensorMutableRawData on a GPU backed OrtValue gives a System.AccessViolationException. I personally think having a distinction in the type system would be nice.

**Comment by yuslepukhin** [comment]
Do not forget about disposing native resources.

**Comment by snnn** [comment]
This issue has been automatically closed as 'not planned' because it has been marked as 'stale' for more than 30 days without activity. If you believe this is still an issue, please feel free to reopen it.
